app/code/Magento/InventorySales/Block/Adminhtml/Order/View/Tab/SourceSelection.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Block\Adminhtml\Order\View\Tab;
+
+use Magento\Backend\Block\Template;
+use Magento\Backend\Block\Template\Context;
+use Magento\Backend\Block\Widget\Tab\TabInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Registry;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\SourceItemRepositoryInterface;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\InventorySales\Api\ShippingAlgorithmInterface;
+use Magento\InventorySales\Api\SourceSelectionInterface;
+
+/**
+ * Tab for source items display on the order editing page
+ *
+ * @api
+ */
+class SourceSelection extends Template implements TabInterface
+{
+    /**
+     * @var Registry
+     */
+    private $registry;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var SourceItemRepositoryInterface
+     */
+    private $sourceItemRepository;
+
+    /**
+     * @var SourceRepositoryInterface
+     */
+    private $sourceRepository;
+    /**
+     * @var ShippingAlgorithmInterface
+     */
+    private $shippingAlgorithm;
+
+    /**
+     * @param Context $context
+     * @param Registry $registry
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param SourceItemRepositoryInterface $sourceItemRepository
+     * @param SourceRepositoryInterface $sourceRepository
+     * @param ShippingAlgorithmInterface $shippingAlgorithm
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        Registry $registry,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        SourceItemRepositoryInterface $sourceItemRepository,
+        SourceRepositoryInterface $sourceRepository,
+        ShippingAlgorithmInterface $shippingAlgorithm,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->registry = $registry;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->sourceItemRepository = $sourceItemRepository;
+        $this->sourceRepository = $sourceRepository;
+        $this->shippingAlgorithm = $shippingAlgorithm;
+    }
+
+    /**
+     * Getting source items for order
+     *
+     * @return array
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getSourceItemsData(): array
+    {
+        $order = $this->registry->registry('current_order');
+
+        $sourceSelectionMap = $this->shippingAlgorithm->get($order)
+                                                    ->getSourceSelections();
+
+        $sourceItemsData = [];
+        foreach ($order->getAllVisibleItems() as $orderItem) {
```

Comment:
Looks like we try to convert ResultObject in raw array data
But is better to avoid this convertation and work with Result object
 
In this case we can avoid double performing of `getSourceItemsBySku` (in block and in algorithm)